pathway-analysis/qusage_meta_analysis.Rmd

```diff
@@ -0,0 +1,429 @@
+    TRUE ~ "SHH"
+  ))
+```
+
+Reorder expression data to match metadata.
+
+```{r}
+robinson_mat <- robinson_mat[, robinson_metadata_df$refinebio_accession_code]
+```
+
+## Pathway analysis
+
+### Read in KEGG pathways
+
+```{r}
+kegg_pathways <- read.gmt(kegg_file)
+```
+
+### Northcott et al.
+
+```{r}
+northcott_results <- qusage(eset = northcott_mat,
+                            labels = northcott_metadata_df$shh_v_other,
+                            contrast = "SHH-Other",
+                            geneSets = kegg_pathways)
+```
+
+Save the Northcott et al. results to file.
+
+```{r}
+northcott_results_file <- file.path("results", 
+                                    "Northcott_SHH-Other_QSarray.RDS")
+readr::write_rds(northcott_results, northcott_results_file)
+```
+
+### Robinson et al.
+
+```{r}
+robinson_results <- qusage(eset = robinson_mat,
+                           labels = robinson_metadata_df$shh_v_other,
+                           contrast = "SHH-Other",
+                           geneSets = kegg_pathways)
+```
+
+Save the Robinson et al. results to file.
+
+```{r}
+robinson_results_file <- file.path("results", 
+                                   "Robinson_SHH-Other_QSarray.RDS")
+readr::write_rds(robinson_results, robinson_results_file)
+```
+
+## Meta-analysis
+
+### Combine probability density function
+
+`combinePDFs` is a `qusage` function we can use for meta-analysis.
+This function accepts a list of `QSArray` results.
+
+```{r}
+results_list <- list(
+  Northcott = northcott_results,
+  Robinson = robinson_results
+)
+```
+
+Because there are more samples in the Northcott et al. dataset, it will be 
+weighted more highly when combining the distribution.
+
+```{r}
+combined_results <- combinePDFs(results_list)
+```
+
+Just as in a single dataset, we can extract relevant information from 
+the combined `QSArray` with built-in `qusage` functions.
+
+Let's look at the top 20 pathways with `qsTable`.
+
+```{r}
+qsTable(combined_results)
+```
+
+### Plotting
+
+We can plot the distributions with `plotCombinedPDF`.
+First let's plot KEGG Pathways in Cancer, which has elevated expression in
+the SHH group as it has a _positive_ fold change.
+
+```{r}
+plotCombinedPDF(combined_results, path.index = 162)
+legend("topleft", legend = c("Northcott", "Robinson", "Meta-analysis"),
+       lty = 1, col = c("#E41A1C", "#377EB8", "black"))
+```
+
+Let's plot KEGG Dorsoventral Axis Formation, which has higher expression in
+Group 3 and 4 samples.
+
+```{r}
+plotCombinedPDF(combined_results, path.index = 107)
+legend("topleft", legend = c("Northcott", "Robinson", "Meta-analysis"),
+       lty = 1, col = c("#E41A1C", "#377EB8", "black"))
+```
+
+#### Plot individual gene mean and 95% confidence intervals
+
+We know that the directionality of the KEGG Dorsoventral Axis Formation pathway
+agrees between datasets.
+We can look into what genes are driving the pathway activity with the 
+`plotCIsGenes` function.
```

Comment:
In these plots, can you explain what the horizontal dashed blue line represents and also what the significance of the darkness of the vertical lines is?

Reply:
Used the option to drop the vertical lines -- don't think the shade means anything